### PR TITLE
[coverage] Various tests (tls, openflow)

### PR DESCRIPTION
**Pull request by gpotter2**
Notes:
- Openflow autocompletion is now enabled by default. This means that a few  tests based on the `raw(s) == ...` format have been changed.
- Fixes some Python3 compat bugs (str/raw)
- Changed the way `prereq_autocompletion` was specified, to follow the same way than the can layer, for instance. Indeed, because of how its imported, it was not possible to set it to True before (its used into the Class defaults before being editable)

### CODE REVIEW

**Comment by gpotter2** [comment]
@guedou This should give us a nice coverage boost

File: scapy/contrib/openflow.py [hunk 1]
```diff
@@ -16,11 +16,11 @@
 from scapy.fields import *
 from scapy.layers.l2 import *
 from scapy.layers.inet import *
-from scapy.compat import orb
+from scapy.compat import orb, raw
 
 ### If prereq_autocomplete is True then match prerequisites will be
 ### automatically handled. See OFPMatch class.
-prereq_autocomplete = False
+conf.contribs['OPENFLOW'] = {'prereq_autocomplete': True}
```
Comment (guedou) [inline scapy/contrib/openflow.py]: Why did you change the default behavior?

File: scapy/contrib/openflow3.py [hunk 2]
```diff
@@ -20,7 +20,7 @@
 
 ### If prereq_autocomplete is True then match prerequisites will be
 ### automatically handled. See OFPMatch class.
-prereq_autocomplete = False
+conf.contribs['OPENFLOW'] = {'prereq_autocomplete': True}
```
Comment (guedou) [inline scapy/contrib/openflow3.py]: Same comment here.

File: scapy/contrib/openflow.py (hunk 1 above)
Comment (gpotter2) [reply scapy/contrib/openflow.py]: Most people won't look at the source code, and will want those packets to be correctly built.
People that really want to send mis-formed packets will notice this
